src/win/i2c.cc

```diff
@@ -0,0 +1,318 @@
+    if (nullptr == obj->_i2cDevice) {
+      Nan::ThrowError(Nan::ErrnoException(EPERM, "WinI2c::SetDevice", "I2cDevice::FromIdAsync failed to return an i2c device"));
+    }
+  }
+}
+
+NAN_METHOD(WinI2c::CloseSync) {
+  WinI2c* obj = Nan::ObjectWrap::Unwrap<WinI2c>(info.This());
+
+  //From https://msdn.microsoft.com/en-us/library/windows.devices.i2c.i2cdevice.close.aspx
+  // You cannot call Close methods through Visual C++ component extensions(C++ / CX) on 
+  // Windows Runtime class instances where the class implemented IClosable.Instead, C++ / CX code 
+  // for runtime classes should call the destructor or set the last reference to null.
+  obj->_i2cDevice = nullptr;
+}
+
+NAN_METHOD(WinI2c::Read) {
+  Nan::ThrowError(Nan::ErrnoException(EPERM, "WinI2c::Read", "Not Implemented"));
+}
+
+NAN_METHOD(WinI2c::ReadPartial) {
+  if (info.Length() < 2 ||
+      !info[0]->IsInt32() ||
+      !info[1]->IsObject()) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "WinI2c::ReadPartial",
+      "incorrect arguments passed to WinI2c::ReadPartial"
+      "(int length, Buffer buffer)"));
+  }
+
+  Nan::HandleScope scope;
+  uint32 length = info[0]->Uint32Value();
+  v8::Local<v8::Object> bufferHandle = info[1].As<v8::Object>();
+  int bytesRead = -1;
+  byte* bufferData = (byte*)node::Buffer::Data(bufferHandle);
+  size_t bufferLength = node::Buffer::Length(bufferHandle);
+
+  if (length > bufferLength) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "WinI2c::ReadPartial",
+      "buffer passed to WinI2c::Read contains less than 'length' bytes"));
+  }
+
+  auto readBuf = ref new Platform::Array<BYTE>(length);
+  WinI2c* obj = Nan::ObjectWrap::Unwrap<WinI2c>(info.This());
+  try {
+    I2cTransferResult result = obj->_i2cDevice->ReadPartial(readBuf);
+
+    switch (result.Status) {
+      case I2cTransferStatus::FullTransfer:
+        bytesRead = result.BytesTransferred;
+        break;
+      case I2cTransferStatus::PartialTransfer:
+        bytesRead = result.BytesTransferred;
+        wprintf(L"%d bytes partially transferred\n", bytesRead);
+        break;
+      case I2cTransferStatus::SlaveAddressNotAcknowledged:
+        wprintf(L"Slave address was not acknowledged\n");
+        break;
+      default:
+        wprintf(L"Invalid transfer status value\n");
+    }
+  }
+  catch (Exception^ e) {
+    Nan::ThrowError(Nan::ErrnoException(e->HResult, "WinI2c::ReadPartial", ""));
+  }
+
+  for (int i = 0; i < readBuf->Length; i++) {
+    bufferData[i] = readBuf[i];
+  }
+
+  if (bytesRead == -1) {
+    return Nan::ThrowError(Nan::ErrnoException(errno, "WinI2c::ReadPartial", ""));
+  }
+
+  info.GetReturnValue().Set(Nan::New<v8::Integer>(bytesRead));
+}
+
+NAN_METHOD(WinI2c::Write) {
+  if (info.Length() < 1 || !info[0]->IsArray()) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "WinI2c::Write",
+      "incorrect arguments passed to WinI2c::Write"));
+  }
+
+  Nan::HandleScope scope;
```

Comment:
An implicit HandleScope is created on JavaScript accessible methods. Why is another one created here?